packages/fxa-settings/src/pages/Signin/SigninRecoveryMethod/Phone/ConfirmRecoveryCode/index.test.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { renderWithLocalizationProvider } from 'fxa-react/lib/test-utils/localizationProvider';
+import { screen, fireEvent, waitFor } from '@testing-library/react';
+import ConfirmRecoveryCode from './index';
+
+describe('ConfirmRecoveryCode', () => {
+  const mockVerifyCode = jest.fn(() => Promise.resolve());
+  const mockResendCode = jest.fn(() => Promise.resolve());
+  const mockClearBanners = jest.fn();
+  const mockSetErrorMessage = jest.fn();
+
+  const defaultProps = {
+    clearBanners: mockClearBanners,
+    maskedPhoneNumber: '••••••1234',
+    errorMessage: '',
+    setErrorMessage: mockSetErrorMessage,
+    verifyCode: mockVerifyCode,
+    resendCode: mockResendCode,
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders as expected', async () => {
+    renderWithLocalizationProvider(<ConfirmRecoveryCode {...defaultProps} />);
+
+    expect(screen.getByRole('button', { name: 'Back' })).toBeVisible();
+    expect(screen.getByRole('heading', { name: 'Sign in' })).toBeVisible();
+    expect(
+      screen.getByRole('heading', { name: 'Enter recovery code' })
+    ).toBeVisible();
+    expect(
+      screen.getByRole('textbox', { name: 'Enter 6-digit code' })
+    ).toBeVisible();
+    expect(screen.getByText('Confirm')).toBeVisible();
+    expect(screen.getByRole('button', { name: 'Resend code' })).toBeVisible();
+    expect(
+      screen.getByRole('link', {
+        name: 'Are you locked out? Opens in new window',
+      })
+    ).toBeVisible();
+  });
+
+  it('submits with valid code', async () => {
+    renderWithLocalizationProvider(<ConfirmRecoveryCode {...defaultProps} />);
+
+    const input = screen.getByRole('textbox');
+    fireEvent.change(input, { target: { value: '123456' } });
```

Comment:
We should use userEvent where possible: https://testing-library.com/docs/user-event/intro/